### ISSUE TITLE
Pod envoy-ratelimit stucks in `Terminating` for a really long time when delete it

### ISSUE THREAD

**Issue by shawnh2** [envoyproxy/gateway#1792]
*Description*:

if you delete the envoy-ratelimit pod, it will stucks in `Terminating` status for a really long time (about at least 5min).

*Repro steps*:

```bash
kubectl delete -n envoy-gateway-system po envoy-ratelimit-7558c5bf58-c4z5v
# wait about 5min, then it returns
pod "envoy-ratelimit-7558c5bf58-c4z5v" deleted

kubectl get po -n envoy-gateway-system -w
NAME                               READY   STATUS    RESTARTS   AGE
envoy-gateway-fbb49f45c-kmhnv      2/2     Running   0          111s
envoy-ratelimit-7558c5bf58-c4z5v   1/1     Running   0          36s
envoy-ratelimit-7558c5bf58-c4z5v   1/1     Terminating   0          51s
envoy-ratelimit-7558c5bf58-ctx27   0/1     Pending       0          0s
envoy-ratelimit-7558c5bf58-ctx27   0/1     Pending       0          0s
envoy-ratelimit-7558c5bf58-ctx27   0/1     ContainerCreating   0          0s
envoy-ratelimit-7558c5bf58-ctx27   1/1     Running             0          1s
envoy-ratelimit-7558c5bf58-c4z5v   0/1     Terminating         0          5m51s
envoy-ratelimit-7558c5bf58-c4z5v   0/1     Terminating         0          5m51s
envoy-ratelimit-7558c5bf58-c4z5v   0/1     Terminating         0          5m51s
envoy-ratelimit-7558c5bf58-c4z5v   0/1     Terminating         0          5m51s  # <= takes about 5min then terminated
```

this issue also happens when apply `helm uninstall`, the envoy-ratelimit pod will spend ~5min to terminate:
```bash
kubectl get po -n envoy-gateway-system
NAME                               READY   STATUS        RESTARTS   AGE
envoy-ratelimit-7558c5bf58-ctx27   1/1     Terminating   0          5m5s
```

*Environment*:

latest

---

*Possible Solutions*:

we could add a field `terminationGracePeriod: 0`(default 30) in ratelimit Pod spec, and this allow kubernetes to send SIGKILL instead of SIGTERM immediately. 

this field works fine just like we are applying:
```bash
kubectl delete -n envoy-gateway-system po envoy-ratelimit-7558c5bf58-lcg8k --grace-period=0
pod "envoy-ratelimit-7558c5bf58-lcg8k" deleted  # return immediately
```

**Comment by arkodg** [comment]
thanks for raising this @shawnh2 , I can see support in `ratelimit` to gracefully handle `SIGTERM` and stop the server https://github.com/envoyproxy/ratelimit/blob/965f0bc861da0d4478071aeca0a5f2c2202257b7/src/server/server_impl.go#L331
can you help further triage this issue, and see why ratelimit is not exiting ?

**Comment by shawnh2** [comment]
it has something to do with this configuration of envoy-ratelimit:

https://github.com/envoyproxy/ratelimit/blob/965f0bc861da0d4478071aeca0a5f2c2202257b7/src/settings/settings.go#L53

the default value of `FORCE_START_WITHOUT_INITIAL_CONFIG` is `false`, so if i enable the envoy-ratelimit without sending the initial config, the server wont start and wont be able to capture the SIGTERM. the server will be blocked on:

https://github.com/envoyproxy/ratelimit/blob/965f0bc861da0d4478071aeca0a5f2c2202257b7/src/service/ratelimit.go#L334

TL;DR **the long duration problem of terminating the envoy-ratelimit pod only happens if the ratelimit-server doesnt receive the initial config.** 

there is no such problem when delete it after sending the initial config.

**Comment by shawnh2** [comment]
possible solutions:

1. set `FORCE_START_WITHOUT_INITIAL_CONFIG` to `true`, this allow ratelimit-server to be able to handle the SIGTERM even without receiving the initial config
2. keep `FORCE_START_WITHOUT_INITIAL_CONFIG` as `false`, just send a empty config from control-plane when successfully connect to envoy-ratelimit config provider

**Comment by shawnh2** [comment]
vote on 1)

they both achieve the same effect, but 1) is the easiest way.

**Comment by shawnh2** [comment]
prefer (1)

(1) and (2) are doing the same thing, but (1) is the easiest way

if we enable ratelimit-server to start without receiving the initial config, not only can it hanlde the SIGTERM, but also get `healthcheck` (for now the envoy-ratelimit pod doesnt have a readiness probe or liveness probe) and `pprof` etc. to work immediately.

**Comment by arkodg** [comment]
1. sounds good @shawnh2 , thanks for debugging this one !
